File: cpp/math/gcd.cpp
using ll = long long;

// Euclidean Algorithm
ll gcd(ll m, ll n) {
  ll r = m % n;
  return r ? gcd(n, r) : n;
}

// Extended Euclidean Algorithm
// - extgcd(a, b, x, y) -> a*x + b*y = d を満たす
ll extgcd(ll a, ll b, ll &x, ll &y) {
  if(b) {
    ll d = extgcd(b, a%b, y, x);
    y -= (a/b)*x;
    return d;
  }
  x = 1; y = 0;
  return a;
}

File: python/math/gcd.py
# Euclidean Algorithm
def gcd(m, n):
    r = m % n
    return gcd(n, r) if r else n

# Extended Euclidean Algorithm
def extgcd(a, b):
    if b:
        d, y, x = extgcd(b, a%b)
        y -= (a/b)*x
        return d, x, y
    else:
        return a, 1, 0

# lcm (least common multiple)
def lcm(m, n):
    return m//gcd(m, n)*n